conn/physic/units_test.go

```diff
@@ -536,3 +538,255 @@ func BenchmarkCelsiusFloatg(b *testing.B) {
 		buf.Reset()
 	}
 }
+
+func TestAtod(t *testing.T) {
+	const (
+		negative = true
+		positive = false
+	)
+	succeeds := []struct {
+		in       string
+		expected decimal
+		n        int
+	}{
+		{"123456789", decimal{"123456789", 0, positive}, 9},
+		{"1nM", decimal{"1", 0, positive}, 1},
+		{"2.2nM", decimal{"22", -1, positive}, 3},
+		{"12.5mA", decimal{"125", -1, positive}, 4},
+		{"-12.5mA", decimal{"125", -1, negative}, 5},
+		{"1ma1", decimal{"1", 0, positive}, 1},
+		{"-0.00001%rH", decimal{"1", -5, negative}, 8},
+		{"0.00001%rH", decimal{"1", -5, positive}, 7},
+		{"--1ma1", decimal{"1", 0, negative}, 3},
```

Comment:
Thinking about it, I don't think these two should be valid.
https://play.golang.org/p/IgZrjBjeoRa

Reply:
agreed, also "-+" and "+-".